refactor: Use C++14 more for a few things

pksunkara opened the pull request:
I gave up on trying to get drafter to compile on xcode

tjanc commented on ext/snowcrash/src/BlueprintParser.h:
```diff
@@ -378,12 +378,9 @@ namespace snowcrash
-
-                        if (!it->symbol.literal.empty() && !it->symbol.variable) {
-                            pd.namedTypeDependencyTable[identifier].insert(it->symbol.literal);
+                    for (auto it : typeDefinition.typeSpecification.nestedTypes) {
```
Consider using `const auto&` rather than `auto`.

tjanc replied:
Not an iterator anymore, consider renaming `it`; for example to `nestedType`.

tjanc commented on ext/snowcrash/src/BlueprintParser.h:
```diff
@@ -410,19 +407,13 @@ namespace snowcrash
-            for (depIt = pd.namedTypeDependencyTable.begin(); depIt != pd.namedTypeDependencyTable.end(); depIt++) {
-
-                resolveNamedTypeDependencyTableEntry(pd, depIt->first, report);
+            for (auto it : pd.namedTypeDependencyTable) {
```
Consider using `const auto&` rather than `auto`.

tjanc replied:
Not an iterator anymore, consider renaming `it`; for example to `dependency`.

tjanc commented on ext/snowcrash/src/BlueprintParser.h:
```diff
@@ -410,19 +407,13 @@ namespace snowcrash
-            for (it = pd.namedTypeInheritanceTable.begin(); it != pd.namedTypeInheritanceTable.end(); it++) {
-
-                resolveNamedTypeBaseTableEntry(pd, it->first, it->second.first, it->second.second, report);
+            for (auto it : pd.namedTypeInheritanceTable) {
```
Consider using `const auto&` rather than `auto`.

tjanc replied:
Not an iterator anymore, consider renaming `it`; for example to `base`.

tjanc commented on ext/snowcrash/src/BlueprintParser.h:
```diff
@@ -614,12 +605,9 @@ namespace snowcrash
-
-                if (it->element == Element::CategoryElement && it->category == Element::ResourceGroupCategory
-                    && it->attributes.name == name) {
+            for (auto it : blueprint.content.elements()) {
```
Consider using `const auto&` rather than `auto`.

tjanc replied:
Not an iterator anymore, consider renaming `it`; for example to `el`.

tjanc commented on ext/snowcrash/src/BlueprintParser.h:
```diff
@@ -678,17 +666,14 @@ namespace snowcrash
-                 ++resourceElementIt) {
-
-                if (resourceElementIt->element == Element::ResourceElement) {
+            for (auto resourceElementIt : element.content.elements()) {
```
Consider using `const auto&` rather than `auto`.

pksunkara replied:
Can't use `const auto&` here. But used `auto&`

tjanc commented on ext/snowcrash/src/BlueprintParser.h:
```diff
@@ -678,17 +666,14 @@ namespace snowcrash
-                 ++resourceElementIt) {
-
-                if (resourceElementIt->element == Element::ResourceElement) {
+            for (auto resourceElementIt : element.content.elements()) {
```
Not an iterator anymore, consider renaming `resourceElementIt`; for example to `resourceElement`.

tjanc commented on ext/snowcrash/src/ResourceGroupParser.h:
```diff
@@ -252,12 +252,9 @@ namespace snowcrash
-
-                if (it->element == Element::CategoryElement
-                    && SectionProcessor<Resource>::isResourceDuplicate(it->content.elements(), uri)) {
+            for (auto it : blueprint.content.elements()) {
```
Consider using `const auto&` rather than `auto`.

tjanc replied:
Not an iterator anymore, consider renaming `resourceElementIt`; for example to `element`.

tjanc commented on ext/snowcrash/src/ResourceGroupParser.h:
```diff
@@ -273,10 +270,8 @@ namespace snowcrash
-            for (Elements::const_iterator it = elements.begin(); it != elements.end(); ++it) {
-
-                if (it->element == Element::ResourceElement) {
+            for (auto it : elements) {
```
Consider using `const auto&` rather than `auto`.

tjanc replied:
Not an iterator anymore, consider renaming `it`; for example to `element`.

tjanc commented on ext/snowcrash/src/ResourceGroupParser.h:
```diff
@@ -291,13 +286,11 @@ namespace snowcrash
-
-                if (it->element == Element::ResourceElement) {
-                    last = it->content.resource;
+            for (auto it : elements) {
```
Consider using `const auto&` rather than `auto`.

tjanc replied:
Not an iterator anymore, consider renaming `it`; for example to `element`.

tjanc commented on ext/snowcrash/src/ResourceParser.h:
```diff
@@ -516,11 +516,8 @@ namespace snowcrash
-
-                if (it->element == Element::ResourceElement && it->content.resource.uriTemplate == uri) {
-
+            for (auto it : elements) {
```
Consider using `const auto&` rather than `auto`.

tjanc replied:
Not an iterator anymore, consider renaming `it`; for example to `element`.

tjanc commented on ext/snowcrash/test/test-MSONPropertyMemberParser.cc:
```diff
@@ -623,7 +623,7 @@ TEST_CASE("Parse mson property containing list of value with string type specifi
 }
 
-TEST_CASE("Parse fixed type attribute on array with primitive nested type", "[mson][property_member][now]")
+TEST_CASE("Parse fixed type attribute on array with primitive nested type", "[mson][property_member]")
```
Why was the `[now]` tag removed?

pksunkara replied:
That tag was supposed to be temporary for testing a single test and I forgot to remove it before

tjanc commented on ext/snowcrash/test/test-ResourceParser.cc:
```diff
@@ -488,7 +488,7 @@ TEST_CASE("Parse model with unrecognised resource", "[resource][model]")
 }
 
-TEST_CASE("Parse named resource with lazy referencing", "[resource][model][issue][84][now]")
+TEST_CASE("Parse named resource with lazy referencing", "[resource][model][issue][84]")
```
Why was the `[now]` tag removed?

pksunkara replied:
That tag was supposed to be temporary for testing a single test and I forgot to remove it before

pksunkara commented:
Addressed the review. There are more for loops which could be improved. But the iterators of those loops are being dereferenced inside them. I needed to do some research before fixing them which is why I didn't do it at this time.
